k8s/event-leader-lease.yaml

```diff
@@ -0,0 +1,34 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: logdna-agent-reporter-lease
+  namespace: logdna-agent
+rules:
+- apiGroups: ["coordination.k8s.io"]
+  resources: ["leases"]
+  verbs: ["get", "watch", "list", "update", "patch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: logdna-agent-k8-events-lease
+  namespace: logdna-agent
+subjects:
+- kind: ServiceAccount
+  name: logdna-agent
+  namespace: logdna-agent
+roleRef:
+  kind: Role
+  name: logdna-agent-k8-events-lease
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: coordination.k8s.io/v1
+kind: Lease
+metadata:
+  name: logdna-agent-k8-events-lease
```

Comment:
Looks like you've switched naming convention in the rest of the PR from `k8-events` -> `reporter`, should probably change here as well?

Reply:
actually two seperate features, kind of confusing.

reporter is kind of reporting on metrics
events is the k8 eventrs ie create, delete, etc...